### PR TITLE
Added iterator proxies, first part. Added first DiSk++ tutorial.

### PR DESCRIPTION
**Pull request by datafl4sh**
Added iterator proxies in order to iterate with
  `for (auto& cl : cells(msh)) {}`
and
  `for (auto& fc : faces(msh)) {}`

Added a tutorial to show how to iterate over mesh elements.

### CODE REVIEW

**Comment by npaster** [comment]
This is ok for me. It will be really easier to iterate on faces. Maybe you can add the same thing for the boundary faces.

File: apps/diskpp_tutorials/diskpp_tutorial_1.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return 1;
+    }
+
+    using T = double;
+    disk::cartesian_mesh<T, 2> msh;
+
+    /* Load the mesh */
+    bool success = disk::load_cartesian_2d_mesh<T>(argv[1], msh);
+    if (!success)
+        return;
```
Comment (npaster) [inline apps/diskpp_tutorials/diskpp_tutorial_1.cpp:48]: I have this problem when I compile. If I return 0 for example, it works

/Users/npignet/Documents/Diskpp/diskpp/apps/diskpp_tutorials/diskpp_tutorial_1.cpp:48:9: error: non-void function 'main' should return a value [-Wreturn-type]
        return;

**Comment by datafl4sh** [comment]
> This is ok for me. It will be really easier to iterate on faces. Maybe you can add the same thing for the boundary faces.

It will arrive soon :) I am reworking the whole iteration stuff and preparing a prototype of an update where we get rid of all the `O(logn)` lookups.

File: apps/diskpp_tutorials/diskpp_tutorial_1.cpp (hunk 1 above)
Comment (datafl4sh) [reply apps/diskpp_tutorials/diskpp_tutorial_1.cpp:48]: OK, my bad. I'll fix it right now.

**Pull request merged**